torchgeo/datasets/levircd.py

```diff
@@ -0,0 +1,190 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""LEVIR-CD+ dataset."""
+
+import glob
+import os
+from typing import Callable, Dict, Optional
+
+import numpy as np
+import torch
+from PIL import Image
+from torch import Tensor
+
+from .geo import VisionDataset
+from .utils import download_and_extract_archive
+
+
+class LEVIRCDPlus(VisionDataset):
+    """LEVIR-CD+ dataset.
+    The `LEVIR-CD+ <https://github.com/S2Looking/Dataset>`_
+    dataset is a dataset for building change detection.
+
+    Dataset features:
+    * image pairs of 20 different urban regions across Texas between 2002-2020
+    * binary change masks representing building change
+    * three spectral bands - RGB
+    * 985 image pairs with 50 cm per pixel resolution (~1024x1024 px)
+
+    Dataset format:
+    * images are three-channel pngs
+    * masks are single-channel pngs where no change = 0, change = 255
+
+    Dataset classes:
+    1. no change
+    2. change
+
+    If you use this dataset in your research, please cite the following paper:
+    * https://arxiv.org/abs/2107.09244
+    """  # noqa: E501
+
+    url = "https://drive.google.com/file/d/1JamSsxiytXdzAIk6VDVWfc-OsX-81U81/view?usp=sharing"
```

Comment:
If you remove `/view?usp=sharing` does that help with the download? The `torchgeo.datasets.nwpu.VHR10` dataset downloads from Google Drive successfully so I don't _think_ anything is broken.

Reply:
I actually just installed torchgeo into a Colab instance and tried instantiating the VHR10 dataset with `download=True` and it gave me the same issue where the file is empty so it throws an error for not being a .rar file. Not sure what's going on.

Reply:
Huh, you're right, I get the same thing. Looks like things must have broken. I'll submit a PR to try to fix VHR10 and if it works I'll let you know what to change.

Reply:
Looks like it's a known bug in torchvision: https://github.com/pytorch/vision/issues/4108

I think for now the best strategy is to ignore the issue and hope that torchvision cuts a new release soon. Users who need to use LEVIR-CD+ or VHR-10 can simply download and extract the file themselves until then.